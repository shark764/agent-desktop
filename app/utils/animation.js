/**
 * throttle max function call rate to browser refresh rate
 */

const requestAnimationFrame =
  window.requestAnimationFrame ||
  window.webkitRequestAnimationFrame ||
  window.mozRequestAnimationFrame ||
  ((callback) => window.setTimeout(callback, 1000 / 60));

function throttler(...callbackArgs) {
  if (!this.running) {
    this.running = true;
    requestAnimationFrame(() => {
      this.callback(...callbackArgs);
      this.running = false;
    });
  }
}

export function throttleDecorator(funcToThrottle) {
  return throttler.bind({ callback: funcToThrottle, running: false });
}
