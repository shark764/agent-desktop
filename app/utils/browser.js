/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

// Check if browser used is IE11 or greater
// Returns undefined for Chrome|Firefox|Edge
export const isIeEleven = () => document.documentMode && document.documentMode >= 11;
