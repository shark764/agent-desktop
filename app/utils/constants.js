/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

export const colors = {
  lightGray: '#ebebeb',
  offWhite: '#e4e4e4',
  lightRed: '#FE4565',
  regularRed: '#E43D5A',
  darkRed: '#CB3750',
  darkGreen: '#001b1e',
  hoverDarkGreen: 'rgba(0, 27, 30, 0.85)',
  black: '#000',
  white: '#FFFFFF',
  skylightBlue: '#23cdf4',
  toggleChecked: '#54B84F',
  toggleCheckedDisabled: '#b7e3b3',
};
