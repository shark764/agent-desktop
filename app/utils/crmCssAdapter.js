/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
  * crmCssAdapter
  *
  * Updates CSS as needed to work with various CRM integrations
  *
  */

export default function crmCssAdapter (styleObject, styleNameArray, crmModule) {
  let styleAffix;
  if (crmModule === 'salesforce-lightning') {
    styleAffix = 'SfLightning';
  }

  styleNameArray.forEach((styleName) => {
    Object.assign(
      styleObject[styleName],
      styleObject[`${styleName}${styleAffix}`]
    );
  });
}