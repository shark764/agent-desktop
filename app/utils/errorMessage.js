import errorMessages from 'containers/Errors/messages';

export const generateErrorMessage = (errorObject, formatMessage) => {
  let errorDescriptionMessage;
  const { code, interactionFatal } = errorObject;
  if (interactionFatal) {
    errorDescriptionMessage = formatMessage(errorMessages.interactionFailed);
  } else if (code && errorMessages[code]) {
    // Specific error message is found for this code
    errorDescriptionMessage = formatMessage(errorMessages[code]);
  } else if (errorObject.message) {
    // Fallback to message provided by error (for SDK errors)
    errorDescriptionMessage = errorObject.message;
  } else {
    // Use default for other errors (probably made by us)
    errorDescriptionMessage = errorMessages.default;
  }
  if (code) {
    errorDescriptionMessage += formatMessage(errorMessages.code, { code });
  }
  if (errorObject.data && errorObject.data.errorDescription) {
    errorDescriptionMessage += formatMessage(errorMessages.errorDescription, {
      errorDescription: errorObject.data.errorDescription,
    });
  }
  return errorDescriptionMessage;
};
