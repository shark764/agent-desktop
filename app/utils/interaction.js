/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

export const lastMessageFromInteraction = (interaction) => {
  if (
    interaction.channelType === 'messaging' ||
    interaction.channelType === 'sms'
  ) {
    if (interaction.messageHistory) {
      for (let i = interaction.messageHistory.length - 1; i >= 0; i -= 1) {
        // use the last non-system message
        if (interaction.messageHistory[i].type !== 'system') {
          return interaction.messageHistory[i];
        }
      }
    }
  }
  return undefined;
};
