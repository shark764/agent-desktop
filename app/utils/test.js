import { IntlProvider } from 'react-intl';
import enUS from 'translations/en-US.json';

export const getIntlContext = () => {
  // Construct a new `IntlProvider` instance by passing `props` and
  // `context` as React would, then call `getChildContext()` to get the
  // React Intl API, complete with the `format*()` functions.
  const intlProvider = new IntlProvider({ locale: 'en-US', messages: enUS }, {});
  const { intl } = intlProvider.getChildContext();
  return intl;
};

export default { getIntlContext };
