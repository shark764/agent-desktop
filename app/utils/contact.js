/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { isValidEmail, isPossibleNumber } from 'utils/validator';
import isURL from 'validator/lib/isURL';

export const formatValue = (attributeToValidate, value) => {
  let formattedValue;
  switch (attributeToValidate.type) {
    case 'phone':
      formattedValue = value.replace(/[^0-9+*#]/g, '');
      if (formattedValue.indexOf('+') !== 0 && formattedValue.length > 0) {
        formattedValue = `+${formattedValue}`;
      }
      break;
    case 'boolean':
      if (value === 'false' || value === '') formattedValue = false;
      else formattedValue = !!value;
      break;
    default:
      formattedValue = value;
  }
  return formattedValue;
};

export const getError = (attributeToValidate, value) => {
  let error = false;
  if (
    attributeToValidate.mandatory &&
    attributeToValidate.type !== 'boolean' &&
    (value === undefined || value.trim().length < 1)
  ) {
    error = 'errorRequired';
  } else if (value !== undefined && value.length) {
    switch (attributeToValidate.type) {
      case 'email':
        if (!isValidEmail(value)) {
          error = 'errorEmail';
        }
        break;
      case 'phone':
        try {
          if (!isPossibleNumber(value)) {
            error = 'errorPhone';
          }
        } catch (e) {
          error = 'errorPhone';
        }
        break;
      case 'link':
        if (
          !isURL(value, {
            protocols: ['http', 'https'],
            require_protocol: true,
          })
        ) {
          error = 'errorLink';
        }
        break;
      case 'number':
        if (Number.isNaN(Number(value))) {
          error = 'errorNumber';
        }
        break;
      default:
        break;
    }
  }
  return error;
};

export const getLocaleLabel = (attribute, locale) => {
  if (attribute.label[locale] !== undefined) {
    return attribute.label[locale];
  } else if (attribute.label.all !== undefined) {
    // Special case for 'All'. We have our own local translation for it we use.
    return attribute.label.all;
  } else if (attribute.objectName !== undefined) {
    // Fallback for attributes
    return attribute.objectName;
  } else {
    // Fallback for section headers
    return '';
  }
};

export function createSearchQuery(query, nextPage) {
  const encodedQuery = {};
  Object.keys(query).forEach(queryName => {
    const queryToEncode = query[queryName];
    encodedQuery[queryName] = encodeURIComponent(queryToEncode);
  });
  return {
    query: Object.assign(encodedQuery, {
      page: nextPage !== undefined ? nextPage : 1,
    }),
  };
}
