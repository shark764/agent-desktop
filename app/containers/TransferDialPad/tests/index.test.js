import React from 'react';
import { shallow } from 'enzyme';
import { mockStore } from '../../../utils/test';
import TransferDialPadContainer, {
  TransferDialPad,
  mapStateToProps,
  mapDispatchToProps,
} from '../index';

jest.mock('../selectors', () => ({
  selectDialpadTextValidity: () => 'mock selectDialpadTextValidity',
  selectDialpadText: () => 'mock selectDialpadText',
}));
jest.mock('../../TransferMenu/selectors', () => ({
  selectTransferTabIndex: () => 'mock selectTransferTabIndex',
}));
jest.mock('../actions', () => ({
  updateDialpadText: () => 'mock updateDialpadText',
}));

describe('TransferDialPad component render', () => {
  it('when transferTabIndex is 0 & dialpadText is invalid', () => {
    const component = shallow(
      <TransferDialPad
        transferTabIndex={0}
        dialpadTextValidity={false}
        dialpadText="mockInvalidDialpadText"
        updateDialpadText={() => {}}
        transferFromDialpad={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('when transferTabIndex is 1 & dialpadText is valid', () => {
    const component = shallow(
      <TransferDialPad
        transferTabIndex={1}
        dialpadTextValidity
        dialpadText="mockValidDialpadText"
        updateDialpadText={() => {}}
        transferFromDialpad={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
describe('TransferDialPad component event handlers', () => {
  it('extParticipantTransferButton onClick event', () => {
    const mockTransferFromDialpad = jest.fn(
      () => 'mockValidTransferToDialpadText'
    );
    const rendered = shallow(
      <TransferDialPad
        transferTabIndex={0}
        dialpadTextValidity
        dialpadText="mockValidDialpadText"
        updateDialpadText={() => {}}
        transferFromDialpad={mockTransferFromDialpad}
      />
    );
    rendered.find('#extParticipantTransferButton').simulate('click');
    expect(mockTransferFromDialpad).toMatchSnapshot();
  });
  it('dialpad onEnter event', () => {
    const mockTransferFromDialpad = jest.fn(
      () => 'mockValidTransferToDialpadText'
    );
    const rendered = shallow(
      <TransferDialPad
        transferTabIndex={0}
        dialpadTextValidity
        dialpadText="mockValidDialpadText"
        updateDialpadText={() => {}}
        transferFromDialpad={mockTransferFromDialpad}
      />
    );
    rendered.find('#dialpad').simulate('enter');
    expect(mockTransferFromDialpad).toMatchSnapshot();
  });
});
describe('mapStateToProps', () => {
  it('should map redux-states from selectors to component-props correctly', () => {
    expect(mapStateToProps()).toMatchSnapshot();
  });
});
describe('mapDispatchToProps', () => {
  it('should map actions from action-creators to component-props correctly', () => {
    expect(mapDispatchToProps()).toMatchSnapshot();
  });
});
describe('TransferDialPadButton connected component render', () => {
  it('renders with store', () => {
    shallow(<TransferDialPadContainer store={mockStore} />);
  });
});
