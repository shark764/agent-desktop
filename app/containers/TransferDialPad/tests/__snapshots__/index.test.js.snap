// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferDialPad component event handlers dialpad onEnter event 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "mockValidDialpadText",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": "mockValidTransferToDialpadText",
    },
  ],
}
`;

exports[`TransferDialPad component event handlers extParticipantTransferButton onClick event 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "mockValidDialpadText",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": "mockValidTransferToDialpadText",
    },
  ],
}
`;

exports[`TransferDialPad component render when transferTabIndex is 0 & dialpadText is invalid 1`] = `
<div>
  <styled.div>
    <Connect(ErrorBoundaryComponent)
      active={false}
      dialpadText="mockInvalidDialpadText"
      id="dialpad"
      onEnter={[Function]}
      setDialpadText={[Function]}
      transfer={true}
    />
    <InjectIntl(Button)
      disabled={true}
      id="extParticipantTransferButton"
      onClick={[Function]}
      style={
        Object {
          "display": "block",
          "margin": "24px auto 0",
          "width": "134px",
        }
      }
      text={
        Object {
          "defaultMessage": "Add participant",
          "id": "app.containers.TransferMenu.addParticipant",
        }
      }
      type="primaryBlue"
    />
  </styled.div>
</div>
`;

exports[`TransferDialPad component render when transferTabIndex is 1 & dialpadText is valid 1`] = `
<div>
  <styled.div>
    <Connect(ErrorBoundaryComponent)
      active={false}
      dialpadText="mockValidDialpadText"
      id="dialpad"
      onEnter={[Function]}
      setDialpadText={[Function]}
      transfer={true}
    />
    <InjectIntl(Button)
      disabled={false}
      id="extParticipantTransferButton"
      onClick={[Function]}
      style={
        Object {
          "display": "block",
          "margin": "24px auto 0",
          "width": "134px",
        }
      }
      text={
        Object {
          "defaultMessage": "Transfer",
          "id": "app.containers.TransferMenu.transfer",
        }
      }
      type="primaryBlue"
    />
  </styled.div>
</div>
`;

exports[`mapDispatchToProps should map actions from action-creators to component-props correctly 1`] = `
Object {
  "updateDialpadText": [Function],
}
`;

exports[`mapStateToProps should map redux-states from selectors to component-props correctly 1`] = `
Object {
  "dialpadText": "mock selectDialpadText",
  "dialpadTextValidity": "mock selectDialpadTextValidity",
  "transferTabIndex": "mock selectTransferTabIndex",
}
`;
